src/components/flowchart/flowchart.js

```diff
@@ -570,15 +571,16 @@ export class FlowChart extends Component {
 
     // Determine if the click event occurred on the slice button.
     const isSliceButtonClicked =
-      this.slicedPipelineActionRef.current &&
-      this.slicedPipelineActionRef.current.contains(event.target);
+      this.slicedPipelineActionBarRef.current &&
+      this.slicedPipelineActionBarRef.current.contains(event.target);
 
     // Check if the pipeline is sliced, no slice button is clicked, and no filters are applied
-    if (this.props.slicedPipeline && !isSliceButtonClicked) {
-      this.props.onResetSlicePipeline();
-      this.updateSlicedPipelineState(null, null, []);
-      // To reset URL to current active pipeline when click outside of a node on flowchart
-      this.props.toSelectedPipeline();
+    if (
+      this.props.slicedPipeline &&
+      !isSliceButtonClicked &&
+      !this.props.isSlicingPipelineApplied
```

Comment:
I dont completely understand this check. I guess we are trying to see if we need to reset sliced pipeline on chart click. (Is this for the sliced view or before the sliced view ? )

Could you please explain why we need to check for all the 3 conditions when a chart click is triggered ? Also `isSlicingPipelineApplied` is true when someone clicks on slice button ? or before clicking ?

Reply:
this is specifically for handling clicking outside the node. So we every time you click outside an element on the chart, it will trigger this `handleChartClick`, it will clear everything

We want to have this functionality for slicing view, lets say when you select one element, or 2 element, in the preview mode, when you click outside the elements, it should reset the chart. However we don't want this to happen in the SLICED view, because the only way to exit the sliced view is to click on the RESET button. And we also don't want this to happen when clicking on the action bar. 

But now I look at this, we  don't need to check `this.props.slicedPipeline` anymore. 


`isSlicingPipelineApplied is true when someone clicks on slice button` => this is correct